### PR TITLE
[Az.DataProtection] Adding CmkEnryption parameters to Get-AzDataProtectionBackupVault, New-AzDataProtectionBackupVault and Update-AzDataProtectionBackupVault cmdlets

### PR DESCRIPTION
**Pull request by Karmah24**
## Description

Adding CmkEnryption parameters to Get-AzDataProtectionBackupVault, New-AzDataProtectionBackupVault and Update-AzDataProtectionBackupVault cmdlets

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Nickcandy** [comment]
Please fix the conflicts

File: src/DataProtection/DataProtection.Autorest/README.md [hunk 1]
```diff
@@ -246,7 +250,7 @@ directive:
       subject: BackupVaultOperationResult
     remove: true
   - where:
-      verb: New
+      verb: New|Update
```
Comment (Karmah24) [inline src/DataProtection/DataProtection.Autorest/README.md:253]: hide only update expanded variant

File: src/DataProtection/DataProtection.Autorest/test/CmkEncryptionScenario.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        $userAssignedIdentity = @{$env.TestCmkEncryption.CmkUserAssignedIdentityId = @{}}
+
+        $vault = New-AzDataProtectionBackupVault -SubscriptionId $env.TestCmkEncryption.SubscriptionId -ResourceGroupName $env.TestCmkEncryption.ResourceGroupName -VaultName $env.TestCmkEncryption.VaultName -Location $env.TestCmkEncryption.Location -StorageSetting $storagesetting -IdentityType UserAssigned -UserAssignedIdentity $userAssignedIdentity -CmkEncryptionState Enabled -CmkIdentityType UserAssigned -CmkUserAssignedIdentityId $env.TestCmkEncryption.CmkUserAssignedIdentityId -CmkEncryptionKeyUri $env.TestCmkEncryption.CmkEncryptionKeyUri  -CmkInfrastructureEncryption Enabled
+
+        $vault.EncryptionSetting.State | Should be "Enabled"
+        $vault.EncryptionSetting.CmkInfrastructureEncryption | Should be "Enabled"
+        $vault.EncryptionSetting.CmkIdentity.IdentityType | Should be "UserAssigned"
+        $vault.EncryptionSetting.CmkIdentity.IdentityId | Should be $env.TestCmkEncryption.CmkUserAssignedIdentityId
+        $vault.EncryptionSetting.CmkKeyVaultProperty.KeyUri | Should be $env.TestCmkEncryption.CmkEncryptionKeyUri
+    }
```
Comment (Karmah24) [inline src/DataProtection/DataProtection.Autorest/test/CmkEncryptionScenario.Tests.ps1:38]: delete vault after checks

**Comment by Karmah24** [comment]
> @Karmah24 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="Microsoft"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

**Comment by Karmah24** [comment]
> @Karmah24 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="Microsoft"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

**Comment by Nickcandy** [comment]
Hi @Karmah24  , our code freeze time is PST 6PM Apr 22. If your code targets to this release, please fix them ASAP

**Comment by Karmah24** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Review by Nickcandy — APPROVED** [review]

**Comment by hiaga** [comment]
Added MUA support for DisableVaultImmutability, Restore, Stop-Protection, Suspend-backup, Disable soft delete operations.

**Pull request merged**